:card_file_box: [MODIFY] Product 엔티티 Column 추가

5hseok opened the pull request:
## 📌 연관된 이슈
closed #2 

## 📝 작업 내용
외부 API에서 반환하는 매물 정보에 따라 매물 유형 타입(원룸, 투룸), 매물 판매 타입(월세, 전세), 매물 사진, 매물 계약 여부, 매물 즉시 구매 가능 여부 를 추가함. ERD 역시 수정해뒀음.

## 💬리뷰 요구사항
한번 확인해주고, ERD와 비교해서 크로스 체크 부탁드립니다. 
그리고, 앞으로도 계속 데이터 다룰 것 같은데, 엔티티 관련 변경사항이 또 발생할 수 있으니, 이 브랜치와 PR은 저번에 말하신대로 닫지 않고 계속 두어 천천히 머지하겠습니다.

5hseok committed:
:card_file_box: [FIX] Product 엔티티 Column 추가 #2

외부 API에서 반환하는 매물 정보에 따라 매물 유형 타입(원룸, 투룸), 매물 판매 타입(월세, 전세), 매물 사진, 매물 계약 여부, 매물 즉시 구매 가능 여부 를 추가함.
ERD 역시 수정해뒀음.

dlawjddn commented:
1. 고아객체 방지 -> cascade

one to many 걸 때, 필수는 아니지만 거의 모든 상황에서 cascade 옵션을 줘야해
cascade는 부모 객체가 변경이 있을 때(특히, 삭제) 그것과 연관관계가 있는 객체도 같이 변경해주는건데

- cascade를 적용해야 하는 경우
User -> UserProduct: 사용자 탈퇴하면? 해당 사용자가 설정한 프로덕트는? 고아객체가 됨, 같이 삭제를 해줘야함

- cascade를 무조건 적용하면 안 되는 경우
Tag -> ProductTag: 여러 사용자가 태그가 사용 중인데 한 사용자가 태그를 사용한다고 해서 그 하위 태그까지 모두 삭제를 해야할까? ㄴㄴ 절대 하지만 필요에 따라 해야하는 경우도 존재

2. Tag의 Enum화

이거는 내가 기능명세나 기획을 몰라서 하는 말이긴 한데 월세, 전세와 같은 방에 대한 옵션이라면 '전세', '월세' 를 굳이 엔티티로 만들어야할까?
불필요한 테이블 같아 보이긴 하는데 기획에 따라서 테이블로 작용할 수도 있을 거 같아서 이거는 너희가 판단해서 적용하면 좋을 거 같고 

- enum?
태그 or 카테고리 or 상태 등을 나타낼 때 굉장히 편리함, 불필요한 테이블을 생성하지 않을 수 있음

- 그러면 월세인 방을 조회한다면 어떻게 ?
jpa: 
ex) HouseRepository -> findAllByTag(Tag tag); 이런 식으로 하면 월세 태그를 가진 프로덕트를 모두 조회할 수 있음

query:
ex) HouseRepository -> findAllByTag(Tag tag); (select * from house  h where  h.tag = :tag)
이런 방식으로 충분히 조회가 가능함
어차피 테이블은 많아지는데, 불필요한 테이블을 줄이는게 중요해서 그래

열심히 하는거 보기 좋다! 개추 ㅋㅋ

5hseok commented:
CASCADE 관련 부분 도움됐어요!!! 감사합니다 ☺️
월세, 전세는 엔티티가 아니라 매물 엔티티에 필드로 들어가게 했습니다! 
읽어주셔서 감사해요 ㅋㅋㅋ 아직 초기 단계라 많이 보완해야겠네요

eeddiinn reviewed (approved):
수정된 ERD랑 비교해서 바뀐 부분 확인 됐습니다 👍

5hseok committed:
:card_file_box: [FIX] Product, User 도메인 cascade 및 orphanRemoval 적용 #2

부모, 자식 관계인 테이블들 간 cascade 및 orphanRemoval 적용함. Product와 UserProduct 같은 관계들은 부모 자식 관계가 아니기 때문에, Repository나 Service 단계에서 코드로 로직 구현 필요
